components/openid/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/openid/OpenIDServerAssociationStore.java

```diff
@@ -54,7 +57,16 @@
      * @param privateAssociations if this association store stores private associations
      */
     public OpenIDServerAssociationStore(String associationsType) {
-        storeId = new Random().nextInt(9999);
+        try {
+            SecureRandom secureRandom = SecureRandom.getInstance(SHA_1_PRNG);
+            storeId = secureRandom.nextInt(9999);
+        } catch (NoSuchAlgorithmException e) {
+            // Ignore Exception.
+            // Tha SHA_1_PRNG algorithm is provided.
+            if (log.isDebugEnabled()) {
+                log.debug(e.getMessage(), e);
```

Comment:
Instead of using the e.getMessage() add a meaningful message.
Ex: 'Ignoring NoSuchAlgorithmException'